contracts/standard/ExampleNFT.cdc

```diff
@@ -0,0 +1,376 @@
+/* 
+*
+*  This is an example implementation of a Flow Non-Fungible Token
+*  It is not part of the official standard but it assumed to be
+*  similar to how many NFTs would implement the core functionality.
+*
+*  This contract does not implement any sophisticated classification
+*  system for its NFTs. It defines a simple NFT with minimal metadata.
+*   
+*/
+
+import "NonFungibleToken"
+import "MetadataViews"
+import "ViewResolver"
+import "FungibleToken"
+
+access(all) contract ExampleNFT: ViewResolver {
+
+    access(all) var totalSupply: UInt64
+
+    access(all) event ContractInitialized()
+    access(all) event Withdraw(id: UInt64, from: Address?)
+    access(all) event Deposit(id: UInt64, to: Address?)
+
+    access(all) event CollectionCreated(id: UInt64)
+    access(all) event CollectionDestroyed(id: UInt64)
+
+    access(all) let CollectionStoragePath: StoragePath
+    access(all) let CollectionPublicPath: PublicPath
+    access(all) let MinterStoragePath: StoragePath
+    access(all) let MinterPublicPath: PublicPath
+
+    access(all) resource NFT: NonFungibleToken.NFT, ViewResolver.Resolver {
+        access(all) let id: UInt64
+
+        access(all) let name: String
+        access(all) let description: String
+        access(all) let thumbnail: String
+        access(self) let royalties: [MetadataViews.Royalty]
+
+        init(
+            id: UInt64,
+            name: String,
+            description: String,
+            thumbnail: String,
+            royalties: [MetadataViews.Royalty]
+        ) {
+            self.id = id
+            self.name = name
+            self.description = description
+            self.thumbnail = thumbnail
+            self.royalties = royalties
+        }
+    
+        access(all) view fun getViews(): [Type] {
+            return [
+                Type<MetadataViews.Display>(),
+                Type<MetadataViews.Royalties>(),
+                Type<MetadataViews.Editions>(),
+                Type<MetadataViews.ExternalURL>(),
+                Type<MetadataViews.NFTCollectionData>(),
+                Type<MetadataViews.NFTCollectionDisplay>(),
+                Type<MetadataViews.Serial>()
+            ]
+        }
+
+        access(all) view fun resolveView(_ view: Type): AnyStruct? {
```

Comment:
this is not view anymore, so if you want to you can remove it, but it is not a big deal